### DISCUSSION TITLE
Reading Shoutcast metadata from source and mapping to icecast output

### DISCUSSION THREAD

**Discussion by DJBenson** [savonet/liquidsoap#3305] [User Support]
I have just set up Icecast and liquidsoap on a server to live stream from my Traktor DJ controller. The client looks like it streams in Shoutcast format (I can connect to the harbor server but not directly to the Icecast server which I think confirms my suspicions...I can't find online where it specifies what format the controller streams in/to).

I have my liquidsoap (v2.1.4) working fine with the exception of metadata. Apparently the controller should be sending metadata to the Shoutcast server but it's not passed on to the Icecast server (my end clients). I have added the harbor.icy_formats setting and added "audio/ogg" to the list (as this is the only format the controller supports) but the Icecast server only ever shows "unknown" for the current track and none of the defaults from the Traktor software, so again assuming this are not being passed through.

Full config shown below - there are still insecure details used as I am testing internally.

```
# Serveur settings
settings.init.allow_root.set(true)
settings.harbor.bind_addrs.set(["0.0.0.0"])
settings.harbor.icy_formats.set(["audio/ogg", "audio/mpeg"])

# Simulcast
simulcast = input.http("http://icecast.thisisdax.com/HeartDanceMP3")


offline = single("/home/offline.mp3")

# A live source
live = input.harbor("live",port=8080,password="hackme")

# fallback
radio = fallback(track_sensitive=false, [live,simulcast,offline])

# mp3@128k output
output.icecast(
  %ffmpeg(format="mp3", %audio(codec="libmp3lame",samplerate=44100,b="128k")),
  mount="live.mp3",
  name="Ribble Dance",
  description=":: Ribble Dance :: Currently Simulcasting Heart Dance ::",
  genre="Dance",
  host="127.0.0.1",
  port=8000,
  radio)

# mp3@64k output
output.icecast(
  %ffmpeg(format="mp3", %audio(codec="libmp3lame",samplerate=22050,b="64k")),
  mount="live_low.mp3",
  name="Ribble Dance",
  description=":: Ribble Dance :: Currently Simulcasting Heart Dance ::",
  genre="Dance",
  host="127.0.0.1",
  port=8000,
  radio)
```

**Comment by DJBenson** [comment]
I've updated the title as it seems the Traktor controller exposes itself as a v1 icecast server (which from what I'm reading is fine for harbor) but I'd still like to pass through the metadata from client to server to the end client.

**Comment by toots** [comment]
Hi @DJBenson. I'm assuming you mean clients connecting to `input.harbor`.

I would make sure that the ogg decoder has a higher priority than ffmpeg:
```
settings.decoder.priorities.ogg := 15
```

Using this and a simple test script I am able to confirm that we are indeed reading the metadata coming from a client.
